lib/feedsme/index.js

```diff
@@ -372,66 +472,95 @@ Feedsme.prototype.build = function build(data, done) {
  *
  * @param {String} env Environment for the builds.
  * @param {Object} data Package.json of the latest build.
- * @param {Function} done Completion callback
  * @api private
  */
-Feedsme.prototype.resolve = function resolve(env, data, done) {
-  var feedsme = this;
-  var app = feedsme.app;
-  var Dependent = app.models.Dependent;
-  var Package = app.models.Package;
+Feedsme.prototype.resolve = async function resolve(env, data) {
+  const Package = this.models.Package;
+  const opts = { concurrency: this.conc };
 
-  app.log.debug(`Resolving dependent for ${ data.name }`);
+  this.log.debug(`Resolving dependent for ${ data.name }`);
 
   //
   // Only find the top level components, if we also find the dependencies of
   // these dependencies we will trigger multiple builds as a change in a sub
   // dependency will already trigger a build of the parent module, which
   // should cascade upwards until the last dependent module is triggered.
   //
-  async.filter(Object.keys(data.dependencies || {}), function filter(name, next) {
-    Package.get(name, function exists(err, details) {
-      if (err || !details) return next(false);
-      return next(true);
+  const { dependencies, name } = data;
+  const filterer = async (name) => {
+    try {
+      const pkg = await Package.get(name);
+      if (!pkg) return false;
+      return true;
+    } catch (ex) {
+      return false;
+    }
+  };
+
+  // Filter out dependencies not managed by warehouse
+  const results = await pFilter(dependencies, filterer, opts);
+
+  this.log.debug(`Found ${ results.length } private dependencies for ${ data.name }`);
+
+  const limiter = pLimit(this.concurrency);
+  //
+  // Update dependents and dependent_of models
+  //
+  await Promise.all(
+    results.map(dependency => {
+      return limiter(() => this._resolveOne({ name, dependency }));
+    })
+  );
+
+};
+
+Feedsme.prototype._resolveOne = async function _resolveOne({ name, dependency }) {
```

Comment:
you don't technically need the async here since you aren't using await and don't really need the exception->promise rejection logic here either.